src/main/java/duke/Duke.java

```diff
@@ -0,0 +1,145 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+import java.util.ArrayList;
+
+public class Duke {
+    private static Storage storage;
+    private static TaskList tasks;
+    private Ui ui;
+
+    public Duke(String filePath) {
+        ui = new Ui();
+        storage = new Storage(filePath);
+        try {
+            tasks = new TaskList(storage.load());
+        } catch (DukeException e) {
+            //ui.showLoadingError();
+            tasks = new TaskList();
+        }
+    }
+
+    public void run() {
```

Comment:
Adding a method description to conform to the coding standard would be helpful, perhaps consider describing the logic of the method too, since it's very complex.